generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/FunctionDataFetcher.kt

```diff
@@ -123,8 +124,14 @@ open class FunctionDataFetcher(
      * Once all parameters values are properly converted, this function will be called to run a simple blocking function.
      * If you need to override the exception handling you can override this method.
      */
-    protected open fun runBlockingFunction(parameterValues: Map<KParameter, Any?>): Any? = try {
-        fn.callBy(parameterValues)
+    protected open fun runBlockingFunction(
+        environment: DataFetchingEnvironment,
+        parameterValues: Map<KParameter, Any?>
+    ): Any? = try {
+        val coroutineScope = environment.graphQlContext.getOrDefault(CoroutineScope(EmptyCoroutineContext))
+        runBlocking(coroutineScope.coroutineContext) {
```

Comment:
I do understand the reasoning behind this change, but think about it, running a blocking function in a coroutine could be catastrophic just because you need context propagation, there should be another way to deal with this issue.

Reply:
> I do not know your use case, but this issue requires a different solution, and for now, what I would advice is to mark the resolver/function as suspend, what is blocking inside your resolver ?

Dataloaders don't play nice with suspended functions. Because of that all functions which resolve properties via dataloaders are not suspended and the trace breaks.

Reply:
I understand the concern of running a blocking code in a coroutine but it's not dramatically worse than running it in an arbitrary thread, is it?

What if we supply another coroutine scope specifically for sync functions to avoid blocks on suspended functions?

Reply:
is not blocking code in reality, is CPU bound work.

> but it's not dramatically worse than running it in an arbitrary thread

It could be, however, context switching it could be even worst and expensive, lets give it a try, but maybe make it optional through a configuration or something so we can opt in for the functionality and compare CPU usage (context switching concern)

Reply:
sure! what would be the best place to introduce a configuration parameter for this kind of a thing?